tgz.go

```diff
@@ -67,6 +96,10 @@ func (a *Archive) AddFileByContent(src []byte, dest string) error {
 }
 
 func (a *Archive) Close() {
+	close(a.WriteQueue)
```

Comment:
Closing the channel at this point may lead to unintended panics.

Notice that calling `ArchiveWriter.Close` sends on the channel, and sending on a close channel panics.

Anyway, we shall fix this by not exposing the channel to multiple values (Archive, ArchiveWriter).